### PR TITLE
Fix intermittent engine controller test failures

### PR DESCRIPTION
**Pull request by NabeelAhsen**
### What this PR does

> Closes https://github.com/Shopify/shopify_app/issues/1218

**TL;DR** We've been seeing the following failures intermittently on builds for a few weeks now:

1. `NoMethodError: undefined method 'shopify_app_path' for #<Module:xyz>`
2. `NoMethodError: undefined method 'root_path' for #<Module:xyz>`

On closer investigation, the two Engine controllers that keep failing are `sessions_controllers` and `webhooks_controller`. 

**One fix that appears to work is to reload the Rails routes between each test.**

---

I'm not 100% sure what the order of operations are for engines when they mount, but I believe that these two controllers have difficulty finding the following routes mounted on the dummy app between tests:

https://github.com/Shopify/shopify_app/blob/129cea927861a21be2f5d22f6872fc1ff3658667/test/dummy/config/routes.rb#L1-L5

The routes above correspond to the dummy app paths `shopify_app_path` and `root_path` respectively. In Rails 6.x.y, the primary point of failure originates from [`ActionDispatch::Routing::Mapper`](https://www.rubydoc.info/docs/rails/4.1.7/ActionDispatch/Routing/Mapper).

### Reviewer's guide to testing

The issue can be reproduced with the following command:

```sh
bundle exec rake test TESTOPTS="--seed=57116
```

To test this change, do the following:

1. Verify that you can run the command above and get intermittent failures on the `master` branch
2. Verify that you can run the command above and get consistently passing tests on this branch: `bug-fix/fix-intermittent-test-failures`

### Things to focus on

1. Is someone able to pinpoint what exactly it is about `Rails::Engine` that is non-deterministic? I don't think I fully understand why there was sudden inconsistent behaviour when testing engine-namespaced controllers (like `SessionsController` and `WebhooksController`)
2. Are there any other controllers defined by `ShopifyApp` engine that fail intermittently?

### Checklist

Before submitting the PR, please consider if any of the following are needed:

- [x] Update `CHANGELOG.md` if the changes would impact users
- [x] Update `README.md`, if appropriate.
- [x] Update any relevant pages in `/docs`, if necessary
- [x] For security fixes, the [Disclosure Policy](https://github.com/Shopify/shopify_app/blob/master/SECURITY.md#disclosure-policy) must be followed.

### CODE REVIEW

**Comment by andyw8** [comment]
> Is someone able to pinpoint what exactly it is about Rails::Engine that is non-deterministic? I don't think I fully understand why there was sudden inconsistent behaviour when testing engine-namespaced controllers (like SessionsController and WebhooksController)
> Are there any other controllers defined by ShopifyApp engine that fail intermittently?

Even if we can't answer these yet, I think it's still worth shipping this.

**Review by andyw8 — APPROVED** [review]
